chapters/equations.tex

```diff
@@ -421,7 +421,7 @@ \subsection{reinit}\label{reinit}
 The operator reinitializes \lstinline!x! with \lstinline!expr! at an event instant.
 \lstinline!x! is a \lstinline!Real! variable (or an array of \lstinline!Real! variables) that must be selected as a state (resp., states), i.e.\ \lstinline!reinit! on \lstinline!x! implies \lstinline!stateSelect = StateSelect.always! on \lstinline!x!.
 \lstinline!expr! needs to be type-compatible with \lstinline!x!.
-\lstinline!reinit! can for the same variable (resp.\ array of variables) only be applied (either as an individual variable or as part of an array of variables) in one equation (having \lstinline!reinit! of the same variable in \lstinline!when! and \lstinline!elsewhen! of the same variable is allowed).
+\lstinline!reinit! can for the same variable (resp.\ array of variables) only be applied (either as an individual variable or as part of an array of variables) in one equation (having \lstinline!reinit! of a variable in several \lstinline!when!- or \lstinline!elsewhen!-clauses of same \lstinline!when!-statement is allowed).
```

Comment:
I think the misunderstanding comes from `equation` referring to an entire `when`-equation; not one of the sub-equations contained in one of its clauses.  So the point is to not allow this:
```
when foo() then
  reinit(x, 0);
end when;
when bar() then
  reinit(x, 0);
end when;
```

Reply:
Hence, I think this could be a better clarification (also changing _statement_ to _equation_):
```suggestion
\lstinline!reinit! can for the same variable (resp.\ array of variables) only be applied (either as an individual variable or as part of an array of variables) in one \lstinline!when!-equation (having \lstinline!reinit! of a variable in several \lstinline!when!- or \lstinline!elsewhen!-clauses of same \lstinline!when!-equation is allowed).
```

Reply:
> I think the misunderstanding comes from `equation` referring to an entire `when`-equation; not one of the sub-equations contained in one of its clauses.  So the point is to not allow this:

I thought the point was also to not allow this:
```
when foo() then
  reinit(x, 0);
  ...
  reinit(x,1);
end when;
```

Reply:
> Hence, I think this could be a better clarification (also changing _statement_ to _equation_):

Whatever formulation we end up with, can we please change "reinit can for the same variable"? I don't know where "for the same variable part" belongs in that sentence. Is it part of the subject or the verb? Then it should be either "reinit for the same variable can" or "reinit can be applied for the same variable". Also "the same" assumes we talked about this variable before. It should be "a". I can't even get to the technical meaning of this sentence because it takes so much effort to unravel its grammatical structure.

Reply:
OK.  Reformulating:
```suggestion
For any given variable (possibly an array variable), \lstinline!reinit! can only be applied (either to an individual variable or to a part of an array variable) in one \lstinline!when!-equation (applying \lstinline!reinit! to a variable in several \lstinline!when!- or \lstinline!elsewhen!-clauses of the same \lstinline!when!-equation is allowed).
```